nautobot/docs/development/apps/migration/ui-component-framework.md

```diff
@@ -0,0 +1,244 @@
+    queryset = Device.objects.all()
+    
+    object_detail_content = ObjectDetailContent(
+        panels=[
+            ObjectFieldsPanel(
+                fields="__all__",
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+            ),
+            ObjectsTablePanel(
+                table_class=RelatedDeviceTable,
+                table_attribute="related_devices",
+                section=SectionChoices.RIGHT_HALF,
+                weight=100,
+            ),
+        ]
+    )
+```
+
+## Migration Steps
+
+### 1. Identify View Components
+First, analyze your existing view to identify its components:
+
+- Object fields display
+- Related object tables
+- Custom field groups
+- Statistics/counts
+- Text content (descriptions, notes)
+- Custom template sections
+
+### 2. Choose Appropriate Panels
+
+Map your existing components to UI Framework panels:
+
+| Current Feature | UI Framework Panel |
+|----------------|-------------------|
+| Basic field display | `ObjectFieldsPanel` |
+| Related object tables | `ObjectsTablePanel` |
+| Key-value data | `KeyValueTablePanel` |
+| Grouped fields | `GroupedKeyValueTablePanel` |
+| Statistics | `StatsPanel` |
+| Markdown/text content | `TextPanel` or `ObjectTextPanel` |
+
+### 3. Convert Views
+
+1. Update your view class:
+```python
+# Before
+class MyDetailView(generic.ObjectView):
+    template_name = 'myapp/detail.html'
+
+# After
+class MyDetailView(ObjectView):
+    object_detail_content = ObjectDetailContent(
+        panels=[
+            # Panel definitions
+        ]
+    )
+```
+
+2. Move context data into panels:
+```python
+# Before
+def get_context_data(self, **kwargs):
+    context = super().get_context_data(**kwargs)
+    context['stats'] = self.get_stats()
+    return context
+
+# After
+StatsPanel(
+    filter_name="device",
+    related_models=[
+        Device,
+        (Circuit, "circuit_terminations__device__in")
+    ],
+    section=SectionChoices.RIGHT_HALF,
+    weight=100,
+)
+```
+
+### 4. Layout Organization
+
+Organize your panels using sections and weights:
+
+```python
+object_detail_content = ObjectDetailContent(
+    panels=[
+        # Left column
+        ObjectFieldsPanel(
+            section=SectionChoices.LEFT_HALF,
+            weight=100,
+        ),
+        
+        # Right column
+        StatsPanel(
+            section=SectionChoices.RIGHT_HALF,
+            weight=100,
+        ),
+        
+        # Full width at bottom
+        ObjectsTablePanel(
+            section=SectionChoices.FULL_WIDTH,
+            weight=200,
+        ),
+    ]
+)
+```
+
+### 5. Custom Content
+
+For custom content that doesn't fit existing panels:
+
+1. Consider using `TextPanel` with custom formatting
+2. Use `KeyValueTablePanel` with transformed data
+3. Create a custom Panel class if needed:
+
+```python
+from nautobot.core.views.panels import Panel
+
+class CustomPanel(Panel):
+    def get_content(self, context):
```

Comment:
`Panel` doesn't have a `get_content()` API - probably this should be `render_body_content()`?

Although this isn't a great example in any case as the below  would be better accomplished by just specifying `body_content_template_path` at init time and implementing a `get_extra_context()` method instead.